### PR TITLE
switching from `cx-Oracle` to `oracledb`

### PR DESCRIPTION
**Pull request by dungdm93**
Recently, Oracle introduce a brand new python driver for its database name [`python-oracledb`](https://github.com/oracle/python-oracledb)
In comparison with `cx_Oracle`, `python-oracledb` **is a Thin driver by default that doesn’t need Oracle Client libraries**. That is a good reason to switch from `cx-Oracle` to `oracledb`

For more information, you can read [the bog post here](https://levelup.gitconnected.com/open-source-python-thin-driver-for-oracle-database-e82aac7ecf5a).

### CODE REVIEW

File: airflow/providers/oracle/hooks/oracle.py [hunk 1]
```diff
@@ -119,51 +117,40 @@ def get_conn(self) -> 'OracleHook':
...
-            conn_config['encoding'] = conn.extra_dejson.get('encoding')
-            # if `encoding` is specific but `nencoding` is not
-            # `nencoding` should use same values as `encoding` to set encoding, inspired by
-            # https://github.com/oracle/python-cx_Oracle/issues/157#issuecomment-371877993
-            if 'nencoding' not in conn.extra_dejson:
-                conn_config['nencoding'] = conn.extra_dejson.get('encoding')
-        if 'nencoding' in conn.extra_dejson:
-            conn_config['nencoding'] = conn.extra_dejson.get('nencoding')
-        if 'threaded' in conn.extra_dejson:
-            conn_config['threaded'] = conn.extra_dejson.get('threaded')
```
Comment (uranusjr) [inline airflow/providers/oracle/hooks/oracle.py:132]: Why are these removed? I would assume at least `encoding` needs to be kept?

**Comment by uranusjr** [comment]
See also #24311. Some coordination is needed…

File: airflow/providers/oracle/hooks/oracle.py (hunk 1 above)
Comment (dungdm93) [reply airflow/providers/oracle/hooks/oracle.py:132]: Because those are deprecated and has no effect:
https://github.com/oracle/python-oracledb/blob/2068d83baba2c18164285e63b9a16e16e5f9b219/src/oracledb/connection.py#L1047-L1050

For `encoding` and `nencoding`, this is always the value "UTF-8".
https://github.com/oracle/python-oracledb/blob/2068d83baba2c18164285e63b9a16e16e5f9b219/src/oracledb/connection.py#L393

**Comment by dungdm93** [comment]
@uranusjr Thanks.
I think I will close this PR to focus on #24311
